libcaf_core/caf/scheduler.cpp

```diff
@@ -0,0 +1,773 @@
+// This file is part of CAF, the C++ Actor Framework. See the file LICENSE in
+// the main distribution directory for license terms and copyright or visit
+// https://github.com/actor-framework/actor-framework/blob/master/LICENSE.
+
+#include "caf/scheduler.hpp"
+
+#include "caf/actor_ostream.hpp"
+#include "caf/actor_system.hpp"
+#include "caf/actor_system_config.hpp"
+#include "caf/blocking_actor.hpp"
+#include "caf/config.hpp"
+#include "caf/defaults.hpp"
+#include "caf/detail/double_ended_queue.hpp"
+#include "caf/detail/thread_safe_actor_clock.hpp"
+#include "caf/logger.hpp"
+#include "caf/scheduled_actor.hpp"
+#include "caf/scoped_actor.hpp"
+#include "caf/send.hpp"
+#include "caf/thread_owner.hpp"
+
+#include <condition_variable>
+#include <fstream>
+#include <ios>
+#include <iostream>
+#include <memory>
+#include <random>
+#include <thread>
+
+#define CONFIG(str_name, var_name)                                             \
```

Comment:
There are multiple implementations, but this seems to be for the work-stealing implementation specifically. In any case, all calls of this macro are in a single function, so a lambda would be better.

Reply:
Calls are in the initializer list so we cant make a lambda here, but I'll inline the calls, doesn't look bad.